### ISSUE TITLE
addurls --key test failures on windows

### ISSUE THREAD

**Issue by kyleam** [datalad/datalad#5213]
[ This is a dedicated issue for the addurls failures in gh-5210. ]

In gh-5202, @adswa managed to get the addurls tests moved from appveyor's `allowed_failures` block to a regular one.  Sadly, once this hit master, tests recently added in my gh-5184 turned appveyor red again.

https://ci.appveyor.com/project/mih/datalad/builds/36677436/job/7t2ou6p9tejh3qvg

@adswa [mentioned][1] that this is due to quoting.  I'm guessing this is specific to batch commands being used here.  @adswa, can you confirm this by reporting if the addurls tests pass with this change?

```diff
diff --git a/datalad/plugin/addurls.py b/datalad/plugin/addurls.py
index 3441e6696a..b448022cbe 100644
--- a/datalad/plugin/addurls.py
+++ b/datalad/plugin/addurls.py
@@ -44,6 +44,7 @@
 from datalad.utils import (
     ensure_list,
     get_suggestions_msg,
+    on_windows,
     unlink,
 )

@@ -783,7 +784,7 @@ def _add_urls(rows, ds, repo, ifexists=None, options=None,
     if by_key:
         # The by_key parameter isn't strictly needed, but it lets us avoid some
         # setup if --key wasn't specified.
-        if repo.fake_dates_enabled:
+        if repo.fake_dates_enabled or on_windows:
             register_url = RegisterUrl(ds, repo)
         else:
             register_url = BatchedRegisterUrl(ds, repo)
```

[1]: https://github.com/datalad/datalad/issues/5210#issuecomment-739707828

**Comment by adswa** [comment]
Sadly, no. If I apply the above patch to the current master, I'm still seeing failures:

```
(base) C:\Users\datalad\repos\datalad>python -m nose -s -v datalad.plugin.tests.test_addurls:TestAddurls.test_addurls_from_key
datalad.plugin.tests.test_addurls.TestAddurls.test_addurls_from_key('MD5-s{size}--{md5sum}', 'MD5') ... ERROR
datalad.plugin.tests.test_addurls.TestAddurls.test_addurls_from_key('MD5E-s{size}--{md5sum}.dat', 'MD5E') ... ERROR
datalad.plugin.tests.test_addurls.TestAddurls.test_addurls_from_key('et:MD5-s{size}--{md5sum}', 'MD5E') ... ERROR
datalad.plugin.tests.test_addurls.TestAddurls.test_addurls_from_key('et:MD5E-s{size}--{md5sum}.dat', 'MD5') ... ERROR
Versions: annexremote=1.2.1 appdirs=1.4.4 boto=2.49.0 cmd:7z=19.00 cmd:annex=8.20201127-ge25d2c445 cmd:bundled-git=UNKNOWN cmd:git=2.29.2.windows.2 cmd:system-git=2.29.2.windows.2 cmd:system-ssh=for exifread=2.1.2 humanize=3.1.0 iso8601=0.1.13 keyring=21.4.0 keyrings.alt=4.0.0 msgpack=1.0.0 mutagen=1.41.1 requests=2.24.0 tqdm=4.50.2 wrapt=1.12.1
Obscure filename: str=b";&%b5{}'.datc" repr=";&%b5{}'.datc"
Encodings: default='utf-8' filesystem='utf-8' locale.prefered='cp1252'
Environment: PATH='C:\\Users\\datalad\\miniconda3;C:\\Users\\datalad\\miniconda3\\Library\\mingw-w64\\bin;C:\\Users\\datalad\\miniconda3\\Library\\usr\\bin;C:\\Users\\datalad\\miniconda3\\Library\\bin;C:\\Users\\datalad\\miniconda3\\Scripts;C:\\Users\\datalad\\miniconda3\\bin;C:\\Users\\datalad\\miniconda3\\condabin;C:\\Program Files\\Docker\\Docker\\Resources\\bin;C:\\Users\\datalad\\miniconda3\\Library\\ImageMagick-7.0.10-Q16-HDRI;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0;C:\\WINDOWS\\System32\\OpenSSH;C:\\Program Files\\Git\\cmd;C:\\Users\\datalad\\miniconda3;C:\\Users\\datalad\\miniconda3\\Library\\mingw-w64\\bin;C:\\Users\\datalad\\miniconda3\\Library\\usr\\bin;C:\\Users\\datalad\\miniconda3\\Library\\bin;C:\\Users\\datalad\\miniconda3\\Scripts;C:\\Users\\datalad\\AppData\\Roaming\\local\\bin;C:\\Users\\datalad\\AppData\\Local\\Microsoft\\WindowsApps;C:\\ProgramFiles\\7-Zip;C:\\Program Files (x86)\\GnuWin32\\bin;.' GIT_CONFIG_PARAMETERS="'init.defaultBranch=dl-test-branch'"

======================================================================
ERROR: datalad.plugin.tests.test_addurls.TestAddurls.test_addurls_from_key('MD5-s{size}--{md5sum}', 'MD5')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\datalad\miniconda3\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Users\datalad\repos\datalad\datalad\tests\utils.py", line 755, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "C:\Users\datalad\repos\datalad\datalad\plugin\tests\test_addurls.py", line 778, in check_addurls_from_key
    ds.addurls(self.json_file, "{url}", "{name}", exclude_autometa="*",
  File "C:\Users\datalad\repos\datalad\datalad\distribution\dataset.py", line 503, in apply_func
    return f(**kwargs)
  File "C:\Users\datalad\repos\datalad\datalad\interface\utils.py", line 482, in eval_func
    return return_func(generator_func)(*args, **kwargs)
  File "C:\Users\datalad\repos\datalad\datalad\interface\utils.py", line 470, in return_func
    results = list(results)
  File "C:\Users\datalad\repos\datalad\datalad\interface\utils.py", line 455, in generator_func
    raise IncompleteResultsError(
datalad.support.exceptions.IncompleteResultsError: Command did not complete successfully. 2 failed:
[{'action': 'addurls',
  'annexkey': 'MD5-s9--3fb7c40c70b0ed19da713bd69ee12014',
  'message': 'registered URL',
  'path': 'C:\\Users\\datalad\\AppData\\Local\\Temp\\datalad_temp_thilejaz\\a',
  'status': 'error',
  'type': 'file'},
 {'action': 'addurls',
  'annexkey': 'MD5-s9--9b72648021b70b8c522642e4490d7ac3',
  'message': 'registered URL',
  'path': 'C:\\Users\\datalad\\AppData\\Local\\Temp\\datalad_temp_thilejaz\\c',
  'status': 'error',
  'type': 'file'}]
-------------------- >> begin captured logging << --------------------
datalad.utils: Level 5: Importing datalad.utils
datalad.utils: DEBUG: Failed to query or got useless SC_ARG_MAX sysconf, will use hardcoded value: module 'os' has no attribute 'sysconf'
datalad.utils: DEBUG: Maximal length of cmdline string (adjusted for safety margin): 16383
datalad.utils: Level 5: Done importing datalad.utils
datalad.cmd: Level 9: Will use git under '' (no adjustments to PATH if empty string)
asyncio: DEBUG: Using proactor: IocpProactor
datalad.cmd: DEBUG: Async run ['git', '--git-dir=', 'config', '-z', '-l', '--show-origin']
datalad.cmd: DEBUG: Launching process ['git', '--git-dir=', 'config', '-z', '-l', '--show-origin']
datalad.cmd: DEBUG: Process 5140 started
datalad.cmd: DEBUG: Waiting for process 5140 to complete
datalad.cmd: Level 5: Read 1366 bytes from 5140[stdout]
datalad.cmd: DEBUG: Process 5140 exited with return code 0
datalad.cmd: Level 8: Finished running ['git', '--git-dir=', 'config', '-z', '-l', '--show-origin'] with status 0
datalad.ui: Level 5: Starting importing ui
datalad.ui.dialog: Level 5: Starting importing ui.dialog
datalad.ui.dialog: Level 5: Done importing ui.dialog
datalad.ui: Level 5: Initiating UI switcher
datalad.ui: DEBUG: UI set to DialogUI(out=<StreamWrapper>)
datalad.ui: Level 5: Done importing ui
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
--------------------- >> end captured logging << ---------------------

======================================================================
ERROR: datalad.plugin.tests.test_addurls.TestAddurls.test_addurls_from_key('MD5E-s{size}--{md5sum}.dat', 'MD5E')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\datalad\miniconda3\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Users\datalad\repos\datalad\datalad\tests\utils.py", line 755, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "C:\Users\datalad\repos\datalad\datalad\plugin\tests\test_addurls.py", line 778, in check_addurls_from_key
    ds.addurls(self.json_file, "{url}", "{name}", exclude_autometa="*",
  File "C:\Users\datalad\repos\datalad\datalad\distribution\dataset.py", line 503, in apply_func
    return f(**kwargs)
  File "C:\Users\datalad\repos\datalad\datalad\interface\utils.py", line 482, in eval_func
    return return_func(generator_func)(*args, **kwargs)
  File "C:\Users\datalad\repos\datalad\datalad\interface\utils.py", line 470, in return_func
    results = list(results)
  File "C:\Users\datalad\repos\datalad\datalad\interface\utils.py", line 455, in generator_func
    raise IncompleteResultsError(
datalad.support.exceptions.IncompleteResultsError: Command did not complete successfully. 2 failed:
[{'action': 'addurls',
  'annexkey': 'MD5E-s9--3fb7c40c70b0ed19da713bd69ee12014.dat',
  'message': 'registered URL',
  'path': 'C:\\Users\\datalad\\AppData\\Local\\Temp\\datalad_temp_rapj8wwn\\a',
  'status': 'error',
  'type': 'file'},
 {'action': 'addurls',
  'annexkey': 'MD5E-s9--9b72648021b70b8c522642e4490d7ac3.dat',
  'message': 'registered URL',
  'path': 'C:\\Users\\datalad\\AppData\\Local\\Temp\\datalad_temp_rapj8wwn\\c',
  'status': 'error',
  'type': 'file'}]
-------------------- >> begin captured logging << --------------------
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
--------------------- >> end captured logging << ---------------------

======================================================================
ERROR: datalad.plugin.tests.test_addurls.TestAddurls.test_addurls_from_key('et:MD5-s{size}--{md5sum}', 'MD5E')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\datalad\miniconda3\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Users\datalad\repos\datalad\datalad\tests\utils.py", line 257, in _wrap_skip_if
    return func(*args, **kwargs)
  File "C:\Users\datalad\repos\datalad\datalad\tests\utils.py", line 755, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "C:\Users\datalad\repos\datalad\datalad\plugin\tests\test_addurls.py", line 778, in check_addurls_from_key
    ds.addurls(self.json_file, "{url}", "{name}", exclude_autometa="*",
  File "C:\Users\datalad\repos\datalad\datalad\distribution\dataset.py", line 503, in apply_func
    return f(**kwargs)
  File "C:\Users\datalad\repos\datalad\datalad\interface\utils.py", line 482, in eval_func
    return return_func(generator_func)(*args, **kwargs)
  File "C:\Users\datalad\repos\datalad\datalad\interface\utils.py", line 470, in return_func
    results = list(results)
  File "C:\Users\datalad\repos\datalad\datalad\interface\utils.py", line 455, in generator_func
    raise IncompleteResultsError(
datalad.support.exceptions.IncompleteResultsError: Command did not complete successfully. 2 failed:
[{'action': 'addurls',
  'annexkey': 'MD5E-s9--3fb7c40c70b0ed19da713bd69ee12014',
  'message': 'registered URL',
  'path': 'C:\\Users\\datalad\\AppData\\Local\\Temp\\datalad_temp_jfcm74un\\a',
  'status': 'error',
  'type': 'file'},
 {'action': 'addurls',
  'annexkey': 'MD5E-s9--9b72648021b70b8c522642e4490d7ac3',
  'message': 'registered URL',
  'path': 'C:\\Users\\datalad\\AppData\\Local\\Temp\\datalad_temp_jfcm74un\\c',
  'status': 'error',
  'type': 'file'}]
-------------------- >> begin captured logging << --------------------
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
asyncio: DEBUG: Using proactor: IocpProactor
--------------------- >> end captured logging << ---------------------

======================================================================
ERROR: datalad.plugin.tests.test_addurls.TestAddurls.test_addurls_from_key('et:MD5E-s{size}--{md5sum}.dat', 'MD5')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\datalad\miniconda3\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Users\datalad\repos\datalad\datalad\tests\utils.py", line 257, in _wrap_skip_if
    return func(*args, **kwargs)
  File "C:\Users\datalad\repos\datalad\datalad\tests\utils.py", line 755, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "C:\Users\datalad\repos\datalad\datalad\plugin\tests\test_addurls.py", line 778, in check_addurls_from_key
    ds.addurls(self.json_file, "{url}", "{name}", exclude_autometa="*",
  File "C:\Users\datalad\repos\datalad\datalad\distribution\dataset.py", line 503, in apply_func
    return f(**kwargs)
  File "C:\Users\datalad\repos\datalad\datalad\interface\utils.py", line 482, in eval_func
    return return_func(generator_func)(*args, **kwargs)
  File "C:\Users\datalad\repos\datalad\datalad\interface\utils.py", line 470, in return_func
    results = list(results)
  File "C:\Users\datalad\repos\datalad\datalad\interface\utils.py", line 455, in generator_func
    raise IncompleteResultsError(
datalad.support.exceptions.IncompleteResultsError: Command did not complete successfully. 2 failed:
[{'action': 'addurls',
  'annexkey': 'MD5-s9--3fb7c40c70b0ed19da713bd69ee12014',
  'message': 'registered URL',
  'path': 'C:\\Users\\datalad\\AppData\\Local\\Temp\\datalad_temp_i4ge06gr\\a',
  'status': 'error',
  'type': 'file'},
 {'action': 'addurls',
  'annexkey': 'MD5-s9--9b72648021b70b8c522642e4490d7ac3',
  'message': 'registered URL',
  'path': 'C:\\Users\\datalad\\AppData\\Local\\Temp\\datalad_temp_i4ge06gr\\c',
  'status': 'error',
  'type': 'file'}]
-------------------- >> begin captured logging << --------------------
```

I'll try to get into a debugger and re-create my quoting observation. IIRC, the placeholder command arguments from ``ds.addurls(self.json_file, "{url}", "{name}", exclude_autometa="*",  key=key_arg)`` in the test got single quoted (I hadn't yet figured out why - I couldn't find a place where quotes would get changed, but I didn't spend too much time on it). I only discovered that running the command manually with double quoting in the debugger worked out nicely.

**Comment by adswa** [comment]
@mih looked a bit into it with me, and found out what is happening. For two out of the three tested files, a key is given. Because there is a key, "git annex fromkey" is called, which fails on Windows (even if symbolic links are enabled). ``RegisterURL`` attaches a very unhelpful error message. This test can not work on Windows.

**Comment by kyleam** [comment]
> Because there is a key, "git annex fromkey" is called, which fails on Windows (even if symbolic links are enabled).

Bah, when initially testing `fromkey` for this, somehow I confused myself into thinking it made a pointer file rather than a symlink when on an adjusted branch.  I should have thought to revisit that assumption given this failure.

> `RegisterURL` attaches a very unhelpful error message.

Indeed, it unintentionally clobbers the error message from `fromkey`.

Thanks, both, for debugging.

I'll work on a fix (at least for git-annex 8.20201116 or later, where `examinekey` has `objectpath` and `objectpointer` fields).
